testing/types.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// Copyright @dsherret and dsherret/conditional-type-checks contributors. All rights reserved. MIT license.
+
+/**
+ * Asserts at compile time that the provided type argument's type resolves to the expected boolean literal type.
+ * @param expectTrue - True if the passed in type argument resolved to true.
+ */
+export function assert<T extends true | false>(expectTrue: T) {
```

Comment:
How about renaming this to `assertType` to avoid confusion with `assert` in `testing/asserts.ts`?